In [6]:
import numpy as np
house_size = np.array([600,750,850,1000,1100,1250,1400,1550,1700,1850])
bedroom= np.array([1,2,2,2,3,3,3,4,4,4])
price= np.array([28,34,38,44,49,55,60,68,73,79])

In [7]:
x= np.column_stack((np.ones(10),house_size,bedroom))
print(x)

[[1.00e+00 6.00e+02 1.00e+00]
 [1.00e+00 7.50e+02 2.00e+00]
 [1.00e+00 8.50e+02 2.00e+00]
 [1.00e+00 1.00e+03 2.00e+00]
 [1.00e+00 1.10e+03 3.00e+00]
 [1.00e+00 1.25e+03 3.00e+00]
 [1.00e+00 1.40e+03 3.00e+00]
 [1.00e+00 1.55e+03 4.00e+00]
 [1.00e+00 1.70e+03 4.00e+00]
 [1.00e+00 1.85e+03 4.00e+00]]


In [8]:
X_transpose_X= x.T @ x
print(X_transpose_X)

[[1.00000e+01 1.20500e+04 2.80000e+01]
 [1.20500e+04 1.60925e+07 3.74500e+04]
 [2.80000e+01 3.74500e+04 8.80000e+01]]


In [9]:
inverse=np.linalg.inv(X_transpose_X)
print(inverse)

[[ 1.02576156e+00 -8.87551711e-04  5.13350884e-02]
 [-8.87551711e-04  7.22075968e-06 -2.79052275e-03]
 [ 5.13350884e-02 -2.79052275e-03  1.18258744e+00]]


In [10]:
X_transpose_Y=x.T @ price
print(X_transpose_Y)

[5.280e+02 7.009e+05 1.632e+03]


In [11]:
beta =inverse @ X_transpose_Y
print(beta)

[3.29597593 0.03827003 1.21022941]


In [12]:
new_house=np.array([1,1300,3])


In [13]:
prediction =new_house @ beta 
print(prediction)

56.67769838285121


## Conclusion

Using the Normal Equation, I trained a linear regression model using house size and number of bedrooms.

For a 1300 sq. ft. house with 3 bedrooms, the model predicted a price of approximately 56.68 lakhs.

In [14]:
import numpy as np
new_house = np.array([1, 1300, 3])

prediction = new_house @ beta

print("Predicted price:", prediction)

Predicted price: 56.67769838285121


In [16]:
test_house = np.array([1, 1000, 2])

test_prediction = test_house @ beta

print("Predicted price:", test_prediction)

Predicted price: 43.98646107559267


## Model Testing

The model was tested with a 1000 sq. ft., 2-bedroom house.

The predicted price was approximately 43.98 lakhs, which is very close to the actual dataset price of 44 lakhs.